src/VisualStudio/Core/Def/TaskList/ProjectExternalErrorReporter.cs

```diff
@@ -152,8 +173,19 @@ private async Task AddNewErrorsAsync(
     }
 
     public int ClearAllErrors()
+        => ClearErrorsWorker();
+
+    public int ClearErrors()
+        => ClearErrorsWorker();
+
+    private int ClearErrorsWorker()
     {
-        DiagnosticProvider.ClearErrors(_projectId);
+        _taskQueue.AddWork(cancellationToken =>
```

Comment:
Isn't there a "cancel existing work" flag you could add here?

Reply:
yup.  moved to that.